### PR TITLE
RSE-1175: Add unit tests

### PR DESCRIPTION
**Pull request by erawat**
## Overview
This PR is to add unit tests to test on existing functions that the new Direct Debit Setup form uses and also fix the warning issue when submit the form without define _dd_ref_ parameter. 

The Test include: 

- Test Fabricators
- RecurrMandateRefTest 
- DirectDebitDataProviderTest 
- MandateStorageManagerTest
- SettingsManagerTest

## Before

- RecurrMandateRef BAO class did not have test functionality.
- DirectDebitDataProvider class did not have test functionality. 
- MandateStorageManager class did not have test functionality. 
- SettingsManager class did not have test functionality. 
- bootstrap.php and manualdirectdebit.civix.php are always checked by phpcs linter
- Warning will be thrown at screen when saving a new mandate.  

## After
- Unit tests can executed on Github Action, during development. 
- bootstrap.php and manualdirectdebit.civix.php will be excluded from phpcs linter.
- mandate entity will be created correctly. 

## Technical Details

Unit test bootstrap requres settings level to ensure that extension settings are activated and can be used when tests run. 

`eval(cv('php:boot --level=settings', 'phpcode'));
`

### CODE REVIEW

File: CRM/ManualDirectDebit/Form/SetUp.php [hunk 1]
```diff
@@ -138,6 +138,12 @@ private function createDirectDebitMandate($values) {
       'label' => "0N",
     ])['values'][0]['value'];
 
+    $originateNumber = civicrm_api3('OptionValue', 'get', [
+      'sequential' => 1,
+      'option_group_id' => 'direct_debit_originator_number',
+      'label' => "01",
```
[the remark this answers is not in the document]
Comment (erawat) [reply CRM/ManualDirectDebit/Form/SetUp.php]: @MiyaNoctem  thanks for spotting this out,  I have reverted the change of this to the work stream branch the spec was not clear about the default value of originate number.

File: CRM/ManualDirectDebit/Test/Fabricator/Mandate.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    return [
+      'bank_name' => 'Lloyds Bank',
+      'account_holder_name' => 'John Doe',
+      'ac_number' => '12345678',
+      'sort_code' => '12-34-56',
+      'dd_code' => 1,
+      'dd_ref' => 'DD Ref',
+      'start_date' => $now->format('Y-m-d H:i:s'),
+      'authorisation_date' => $now->format('Y-m-d H:i:s'),
+      'originator_number' => 1,
```
[the remark this answers is not in the document]
Comment (erawat) [reply CRM/ManualDirectDebit/Test/Fabricator/Mandate.php]: @MiyaNoctem  I have added OriginatorNumber Fabricator.

File: tests/phpunit/CRM/ManualDirectDebit/Common/MandateStorageManagerTest.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  protected $contact;
+
+  /**
+   * setUp test
+   * @throws CiviCRM_API3_Exception
+   */
+  public function setUp() {
+    $this->contact = ContactFabricator::fabricate();
+    $originatorNumber = OriginatorNumberFabricator::fabricate()['values'][0]['value'];
+    print_r($originatorNumber);
```
Comment (MiyaNoctem) [inline tests/phpunit/CRM/ManualDirectDebit/Common/MandateStorageManagerTest.php:34]: Guess you should remove this print_r

**Comment by erawat** [comment]
@MiyaNoctem  I have changed the head branch to master branch as there is no new features in this PR. It only contains the unit tests and fabricator for the old functions / classes.

**Pull request merged**